### ISSUE TITLE
readability-trailing-comma - false positive + default function argument

### ISSUE THREAD

**Issue by Lev275568** [llvm/llvm-project#223646]
LLVM version 23.1.1

`clang-tidy --checks='-*,readability-trailing-comma' test.cpp -- -std=c++26`

Reproducer:

```c++
struct Options
{
    int option1{};
    int option2{};
};

class Class
{
public:
    static void setOptions(Options opts1 = {}, Options opts2 = {})
    {
        (void)opts1;
        (void)opts2;
    }

private:
};

```

Output:

```
test.cpp:11:46: error: initializer list should not have a trailing comma [readability-trailing-comma,-warnings-as-errors]
   11 |     static void setOptions(Options opts1 = {}, Options opts2 = {})
      |                                              ^
```
